python/python_direct/ops.cpp

```diff
@@ -2948,6 +2948,183 @@ tuple[TensorView, TensorView, TensorView]
       py::return_value_policy::reference);
 }
 
+template <
+    class ShapeType,
+    TensorView* (*RandomFuncWithSeed)(
+        const std::vector<Val*>&,
+        Val*,
+        Val*,
+        DataType,
+        Val*,
+        Val*,
+        bool)>
+TensorView* random_dist_op_fn(
+    Val* arg1,
+    Val* arg2,
+    ShapeType generic_new_shape,
+    std::optional<Val*> rng_seed,
+    std::optional<Val*> rng_offset,
+    PrimDataType dtype) {
+  NVF_CHECK(
+      !(rng_seed.has_value() ^ rng_offset.has_value()),
+      "rng_seed and rng_offset must be provided together!");
+  std::vector<Val*> new_shape = SequenceAsVector(generic_new_shape);
+  if (rng_seed.has_value() && rng_offset.has_value()) {
+    return RandomFuncWithSeed(
+        new_shape,
+        arg1,
+        arg2,
+        dtype,
+        rng_seed.value(),
+        rng_offset.value(),
+        /*maybe_symbolic=*/true);
+  } else {
+    return RandomFuncWithSeed(
+        new_shape,
+        arg1,
+        arg2,
+        dtype,
+        /*rng_seed=*/nullptr,
+        /*rng_offset=*/nullptr,
+        /*maybe_symbolic=*/true);
+  }
+}
+
+void bindRandomOps(py::module_& ops) {
+  ops.def(
+      "normal",
+      random_dist_op_fn<py::list, normal>,
+      py::arg("mean"),
+      py::arg("std"),
+      py::arg("shape"),
+      py::kw_only(),
+      py::arg("rng_seed") = py::none(),
+      py::arg("rng_offset") = py::none(),
+      py::arg("dtype") = DataType::Float,
+      R"(
+Create a tensor with normal distribution.
+Parameters
+----------
+mean : Val
+    The mean of the normal distribution.
+std : Val
+    The standard deviation of the normal distribution.
+shape : list or tuple
+    The shape of the tensor.
+rng_seed : Val, optional
+    The seed for the random number generator.
+rng_offset : Val, optional
+    The offset for the random number generator.
+dtype : PrimDataType, optional
+    The data type of the tensor.
+
+Returns
+-------
+TensorView
+The tensor with normal distribution.
+      )",
+      py::return_value_policy::reference);
+  ops.def(
+      "normal",
+      random_dist_op_fn<py::tuple, normal>,
+      py::arg("mean"),
+      py::arg("std"),
+      py::arg("shape"),
+      py::kw_only(),
+      py::arg("rng_seed") = py::none(),
+      py::arg("rng_offset") = py::none(),
+      py::arg("dtype") = DataType::Float,
+      R"(
+Create a tensor with normal distribution.
```

Comment:
naive question, I thought we don't need to duplicate the docstring for overload?

Reply:
Appreciate the catch. It isn't need and I missed it.